src/wp-includes/script-modules.php

```diff
@@ -123,3 +123,53 @@ function wp_dequeue_script_module( string $id ) {
 function wp_deregister_script_module( string $id ) {
 	wp_script_modules()->deregister( $id );
 }
+
+/**
+ * Registers all the default WordPress Script Modules.
+ *
+ * @since 6.7.0
+ */
+function wp_default_script_modules() {
+	$suffix = defined( 'WP_RUN_CORE_TESTS' ) ? '.min' : wp_scripts_get_suffix();
+
+	/*
+	 * Expects multidimensional array like:
+	 *
+	 *     'interactivity/index.min.js' => array('dependencies' => array(…), 'version' => '…'),
+	 *     'interactivity/debug.min.js' => array('dependencies' => array(…), 'version' => '…'),
+	 *     'interactivity-router/index.min.js' => …
+	 */
+	$assets = include ABSPATH . WPINC . "/assets/script-modules-packages{$suffix}.php";
+
+	foreach ( $assets as $file_name => $script_module_data ) {
+		/*
+		 * Build the WordPress Script Module ID from the file name.
+		 * Prepend `@wordpress/` and remove extensions and `/index` if present:
+		 *   - interactivity/index.min.js  => @wordpress/interactivity
+		 *   - interactivity/debug.min.js  => @wordpress/interactivity/debug
+		 *   - block-library/query/view.js => @wordpress/block-library/query/view
```

Comment:
I'm wondering if these paths are the same on Windows. The challenge is that for regular scripts, the path contains only the file name. In this case, it's also folders so let's confirm that there isn't `block-library\query\view.js` or something like that. This would mean `script-modules-packages.min.php` under version control differs between operating systems.

Reply:
Maybe @t-hamano could help with that like in https://github.com/WordPress/gutenberg/pull/65064#issuecomment-2343171586.

After `npm run build`, here's what's expected at `build/wp-includes/js/dist/script-modules`

```
# tree build/wp-includes/js/dist/script-modules
build/wp-includes/js/dist/script-modules
├── block-library
│   ├── file
│   │   ├── view.js
│   │   └── view.min.js
│   ├── image
│   │   ├── view.js
│   │   └── view.min.js
│   ├── navigation
│   │   ├── view.js
│   │   └── view.min.js
│   ├── query
│   │   ├── view.js
│   │   └── view.min.js
│   └── search
│       ├── view.js
│       └── view.min.js
├── interactivity
│   ├── debug.js
│   ├── debug.min.js
│   ├── index.js
│   └── index.min.js
└── interactivity-router
    ├── index.js
    └── index.min.js

9 directories, 16 files
```

Reply:
I tested this PR on a Windows host OS. From my understanding, `build:dev` does not generate a `build` directory, but generates some directories in the `src` directory. On the other hand, the `build` command generates a complete WordPress in the `build` directory.

In both commands, the `script-modules-packages.min.php` file contains only forward slashes.

<details><summary>npm run build:dev</summary>

![build-dev-1](https://github.com/user-attachments/assets/b1fdb787-e8ed-4593-92be-e34b9f7c68af)

![build-dev-2](https://github.com/user-attachments/assets/6dcae403-97cb-43e2-944b-9748c599d604)

</details> 

<details><summary>npm run build</summary>

![build-1](https://github.com/user-attachments/assets/9450baee-479b-4d08-bde0-ea2758d5575e)

![build-2](https://github.com/user-attachments/assets/8231fd3b-e5ab-456b-8d7b-b7a9fee2d3c8)

</details> 

I then replaced the core files in my local WordPress environment on my Windows host OS with the files built in this PR.

The image lightbox works and scripts seem to be enqueued correctly. The navigation block works fine too.

![front-end](https://github.com/user-attachments/assets/e502a716-7dc5-4da2-bd42-d7b98cbc74bf)

Reply:
Thank you 👍 

> In both commands, the script-modules-packages.min.php file contains only forward slashes.

Perfect, so that file should be stable across different systems and working correctly 👌

Reply:
@t-hamano, thank you so much for testing and confirming it works the same on Windows.